src/main/java/org/folio/inventory/consortium/services/ConsortiumServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+              }
+              String centralTenantId = userTenants.getJsonObject(0).getString("centralTenantId");
+              LOGGER.debug("getCentralTenantId:: Found centralTenantId: {}", centralTenantId);
+              return CompletableFuture.completedFuture(centralTenantId);
+            } else {
+              String message = String.format("Error retrieving centralTenantId by tenant id: %s, status code: %s, response message: %s",
+                context.getTenantId(), httpResponse.getStatusCode(), httpResponse.getBody());
+              LOGGER.warn(String.format("getCentralTenantId:: %s", message));
+              return CompletableFuture.failedFuture(new ConsortiumException(message));
+            }
+          }));
+    return Future.fromCompletionStage(completableFuture);
+  }
+
+  public Future<String> getConsortiumId(Context context) {
+    CompletableFuture<String> completableFuture = createOkapiHttpClient(context)
+      .thenCompose(client ->
+        client.get(context.getOkapiLocation() + CONSORTIA_ENDPOINT).toCompletableFuture()
+          .thenCompose(httpResponse -> {
+            if (httpResponse.getStatusCode() == HttpStatus.SC_OK) {
+              JsonArray consortia = httpResponse.getJson().getJsonArray("consortia");
+              if (consortia.isEmpty()) {
+                String message = String.format("ConsortiaId for tenant: %s not found", context.getTenantId());
+                LOGGER.warn(String.format("getConsortiumId:: %s", message));
+                return CompletableFuture.failedFuture(new NotFoundException(message));
+              }
+              String consortiumId = consortia.getJsonObject(0).getString("id");
+              LOGGER.debug("getConsortiumId:: Found consortiumId: {}", consortiumId);
+              return CompletableFuture.completedFuture(consortiumId);
+            } else {
+              String message = String.format("Error retrieving consortiaId by tenant: %s, status code: %s, response message: %s",
+                context.getTenantId(), httpResponse.getStatusCode(), httpResponse.getBody());
+              LOGGER.warn(String.format("getConsortiumId:: %s", message));
+              return CompletableFuture.failedFuture(new ConsortiumException(message));
+            }
+          }));
+    return Future.fromCompletionStage(completableFuture);
+  }
+
+  public Future<SharingInstance> shareInstance(Context context, String consortiumId, SharingInstance sharingInstance) {
+    CompletableFuture<SharingInstance> completableFuture = createOkapiHttpClient(context)
+      .thenCompose(client ->
+        client.post(context.getOkapiLocation() + String.format(SHARE_INSTANCE_ENDPOINT, consortiumId), Json.encode(sharingInstance))
+          .thenCompose(httpResponse -> {
+            if (httpResponse.getStatusCode() == HttpStatus.SC_OK && !httpResponse.getJson().getString("status").equals(Status.ERROR.toString())) {
```

Comment:
is "status" always present in response? Might be better to compare with Status.ERROR

Reply:
If the sharing status is "IN_PROGRESS", this code will assume the sharing is successful. Is my statement correct?

Reply:
@okolawole-ebsco Your statement is correct, in this case, when we are pulling instance from the central tenant sharing status should be "COMPLETE" or "ERROR". https://github.com/folio-org/mod-consortia/blob/master/src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java#L101